src/endstop.c

```diff
@@ -17,18 +18,34 @@ struct endstop {
     uint32_t rest_time, sample_time, nextwake;
     struct trsync *ts;
     uint8_t flags, sample_count, trigger_count, trigger_reason;
+    uint8_t type;
 };
 
 enum { ESF_PIN_HIGH=1<<0, ESF_HOMING=1<<1 };
 
 static uint_fast8_t endstop_oversample_event(struct timer *t);
+uint16_t BD_Data=0;
 
+static uint8_t
+read_endstop_pin(struct endstop *e)
+{
+#if CONFIG_MACH_STM32F031
+    return gpio_in_read(e->pin);
+#else
+    if(e->type==2)// for Bed Distance sensor
+    {
+        return BD_Data?0:1;
+    }
+    else
+        return gpio_in_read(e->pin);
+#endif
+}
```

Comment:
Much like above, I don't think its valid to put probe specific code here.  I'm not sure what the best software solution for this is, I believe that trigger_sync (trsync.c) can be used?  One short term solution would be to jumper two pins, an input pin the the endstop and an output pin your `bd_sensor.c` sets based on the `BD_Data`.